src/infinite-scroll/index.ts

```diff
@@ -0,0 +1,356 @@
+import {
+  alias,
+  configurable,
+  tag,
+  Events,
+  ProductTransformer,
+  Selectors,
+  Store,
+  StoreSections,
+  Tag
+} from '@storefront/core';
+import { Adapters, Routes } from '@storefront/flux-capacitor';
+import { List } from '@storefront/structure';
+
+export const PADDING = 20;
+export const LOADLABEL = 'loading...';
+
+@configurable
+@alias('infinite')
+@tag('gb-infinite-scroll', require('./index.html'), require('./index.css'))
+class InfiniteScroll {
+
+  searchMethods: any = {
+    pageSize: Selectors.pageSize,
+    productsWithMetadata: Selectors.productsWithMetadata,
```

Comment:
One question: it seems in the global state, all products are held in `...data.present.products`, why not use those instead of duplicate the storage in the local state of this component?